File: LayoutTests/http/tests/cache/disk-cache/speculative-validation/resources/css-to-revalidate.php
<?php
if ($_SERVER["HTTP_IF_NONE_MATCH"] == "foo") {
    header("HTTP/1.1 304 Not Modified");
    header("ETag: foo");
    return;
}

header("Content-Type: text/css");
header("ETag: foo");
header("Cache-Control: max-age=0");

?>

.testClass
{
    color: green;
}

File: LayoutTests/http/tests/cache/disk-cache/speculative-validation/resources/css-to-revalidate.py
#!/usr/bin/env python3

import os
import sys

match = os.environ.get('HTTP_IF_NONE_MATCH', '')
if match == 'foo':
    sys.stdout.write(
        'status: 304\r\n'
        'ETag: foo\r\n'
        'Content-Type: text/html\r\n\r\n'
    )
    sys.exit(0)

sys.stdout.write(
    'Content-Type: text/css\r\n'
    'ETag: foo\r\n'
    'Cache-Control: max-age=0\r\n\r\n'
    '.testClass\n'
    '{\n'
    '    color: green;\n'
    '}\n'
)